### PR TITLE
Move all test logic out of travis, into hack/verify-all.sh

### PR DESCRIPTION
**Pull request by fejta**
This will make it trivial to migrate the job to prow

/assign @Katharine @ixdy 

ref https://github.com/kubernetes/test-infra/pull/11987

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,28 +15,6 @@ before_install:
...
-  # TODO(fejta): vendor these things as go modules
-  - go get -u github.com/alecthomas/gometalinter
-  - go get -u github.com/bazelbuild/buildtools/buildifier
-  - gometalinter --install
 
 script:
-  # Build first since we need the generated protobuf for the govet checks
-  - bazel build --config=ci //...
   - ./hack/verify-all.sh
-  - buildifier -mode=check $(find . -name BUILD -o -name '*.bzl' -type f -not -wholename '*/vendor/*')
```
Comment (ixdy) [inline .travis.yml]: it looks like buildifier and gometalinter checks were just removed?

Comment (fejta) [reply .travis.yml]: hack/verify-bazel.sh covers buildifier
nogo does many similar checks to gometalinter (which is getting turned down in a couple weeks)

**Comment by fejta** [comment]
/hold

Will come back to this once I get test-infra building

File: .travis.yml (hunk 1 above)
Comment (ixdy) [reply .travis.yml]: I don't think verify-bazel looks at bzl files though, does it?

**Pull request closed without merging**